share/getters/shrex.go

```diff
@@ -130,6 +130,10 @@ func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.Ex
 	for {
 		if ctx.Err() != nil {
 			sg.metrics.recordEDSAttempt(ctx, attempt, false)
+			if err != nil {
+				// return aggregated error
+				return nil, err
+			}
 			return nil, ctx.Err()
```

Comment:
do you want to just append the context err as well instead of the check?

Reply:
Nice idea! Thank you!